### ISSUE TITLE
ZipWriterPipe cannot close zipentry

### ISSUE THREAD

**Issue by Bibi0803** [frankframework/frankframework#5902]
An adapter using ZipWriterPipe was working with `7.8.1-20230621.190029` but not with `7.8.3-hotfix1`.

ERROR: error processing message with messageId [0ace3815-2838f7cd_18be8c51d78_369b]: : Pipe [writeXmlFileToZipStream] msgId [0ace3815-2838f7cd_18be8c51d78_369b] action [write] zipWriterHandle [zipwriterhandle]cannot add zipentry for [Filename.xml]: cannot close zipentry: (IOException) Stream closed

The adapter hangs there and no Ladybug report generated.

**Reporter**
Bibi

**Configuration**
```
`<pipe name="writeXmlFileToZipStream" 
           className="nl.nn.adapterframework.compression.ZipWriterPipe"
           action="write">
	<param name="filename" 
                      sessionKey="xmlQueryResult"
		      xpathExpression="concat(result/rowset/row/field[@name='NAME'],'.',lower-case(result/rowset/row/field[@name='TYPE']))" />
	<forward name="success" path="storeIntegrityData" />
</pipe>`
```

**Environment**
```
FF! 7.8.3-hotfix1: Ibis4XXXX 17-c2168a0
Running on SRTZZAPP0524 using IBM WebSphere Application Server/9.0.5.17
Java Version: Java(TM) SE Runtime Environment (8.0.8.10 - pwa6480sr8fp10-20230703_02(SR8 FP10))
Heap size: 414M, total JVM memory: 1024M
Free disk space: 73GB, total disk space: 100GB
Up since: 2023-11-17 11:10:37 (13d)
```

**Comment by jkosternl** [comment]
Spoke with Bibi this morning and she prepared a nice Ladybug report/scenario. 
[Pipeline HandleScannedDocumentsOut.zip](https://github.com/frankframework/frankframework/files/13634880/Pipeline.HandleScannedDocumentsOut.zip)

**Comment by jkosternl** [comment]
Update: Bib has tried out the `7.9-RC1-hotfix10` release as an experiment, which is dated at 12/12/2023 on 7.9-branch. That release seems to work with the same pipelines. So apparently, this bug only happens in 7.8 and is introduced between `7.8.1-20230621.190029` and `7.8.3-hotfix1` (date: 28/10/2023) on 7.8-branch.

**Comment by jkosternl** [comment]
Update: today we've investigated this issue, by back porting Larva-tests from 7.9 to the 7.8 branch. While doing this, we found a bug about closing twice a stream, which should not happen. We've fixed that in the `ZipWriterPipe` inside PR #5988. But we are not sure this fixes this issue as reported.

We've also performed some analyses on the pipelines of the given Ladybug report. We have the following finding:
The pipeline `addDocumentFilesToZipStream` uses the session variable `zipwriterhandle` which is fine. But pipeline  `createDocumentStream` uses the same `zipwriterhandle` implicitly, because it is not specified otherwise and uses the default name, which is the same. And later, below that pipeline, the same `zipwriterhandle` is reused again. This is not a good practice. A stream should not be written, read and written again. Perhaps this worked before in 7.8 releases, but this is not how it should be used. Perhaps the pipelin `createDocumentStream` should create a different stream to use, so the original stream `zipwriterhandle` is not touched.
